Refactor wallet connection logic to utilize usePrivyAddress hook

Nathaniyell opened the pull request:
- Replaced direct usage of useConnectWallet in multiple components with usePrivyAddress for address retrieval.
- Updated the connectWallet function to use the login method from the Privy library.
- Ensured consistent address management across Header, useChat, useInitialMessages, and useSuggestions hooks.

Nathaniyell committed:
Refactor wallet connection logic to utilize usePrivyAddress hook

Nathaniyell committed:
Refactor components to replace useConnectWallet with usePrivyAddress hook for improved address management

sweetmantech commented on hooks/useConnectWallet.tsx:
```diff
@@ -1,23 +1,10 @@
 };
 
-export default useConnectWallet;
+export default useConnectWallet;
```
Please delete this hook. Directly use the `usePrivy` hook anytime you need `login`

sweetmantech commented on components/Header/Header.tsx:
```diff
@@ -6,10 +6,12 @@ import useConnectWallet from "@/hooks/useConnectWallet";
   const { push } = useRouter();
-  const { address, connectWallet } = useConnectWallet();
+  const { connectWallet } = useConnectWallet();
+  const address = usePrivyAddress();
```
Please follow our existing standard for exporting variables from a hook.
```suggestion
  const {address} = usePrivyAddress();
```

sweetmantech commented on hooks/usePrivyAddress.tsx:
```diff
@@ -0,0 +1,9 @@
+
+const usePrivyAddress = (): Address | undefined => {
+  const { user } = usePrivy();
+  return user?.wallet?.address ? getAddress(user.wallet.address) : undefined;
```
```suggestion
  return {address: user?.wallet?.address ? getAddress(user.wallet.address) : undefined}
```

Nathaniyell committed:
Refactor wallet connection logic to use usePrivy for improved address handling and simplify code structure. Removed useConnectWallet hook and updated components to utilize the new address format.

Nathaniyell committed:
Refactor usePrivyAddress hook to define a clearer interface for address management

Nathaniyell committed:
Refactor chat route to use dynamic referenceId from request body

- Updated the POST route in chat API to retrieve referenceId from the request body, allowing for more flexible usage instead of a hardcoded value.

sweetmantech commented on app/api/chat/route.ts:
```diff
@@ -10,9 +10,8 @@ export const POST = enhanceRouteHandler(
-    const referenceId = "if1Fg9bo"; // hard coded for demo
-
-    try {    
+    const referenceId = body.accountId || "if1Fg9bo"; 
```
Do we still need the fallback value?
- What happens if we `remove` the fallback value?

Nathaniyell committed:
Update chat route to remove hardcoded referenceId, using dynamic value from request body
